### DISCUSSION TITLE
2fa SMS while user creation

### DISCUSSION THREAD

**Discussion by haseebma** [dasniko/keycloak-2fa-sms-authenticator#7] [Q&A] [answered]
Hi Niko, You explained very well this topic. Can you please tell me how I can use 2fa SMS while creating user/user registration form?

Thanks

**Comment by dasniko** [comment, marked as answer]
Thanks.

Extending the register flow is a bit exhaustive and awkward. :(
The registration doesn't provide a multi-page flow, you'll have to "fake" it with setting some context values and acting on them in the base .ftl to load/include your desired step-ftl. Then you have to implement some custom form actions (the interfaces for the registration actions are different to the ones of authentication!= to handle all this stuff. Too much stuff to explain it here in short, unfortunately.

**Reply by haseebma** [reply]
Thanks for the explanation.
